### PR TITLE
Sanitize elemental-operator dependencies

### PR DESCRIPTION
**Pull request by anmazzotti**
Part of https://github.com/rancher/elemental-operator/issues/672

### CODE REVIEW

File: cmd/operator/main.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ package main
 
 import (
 	"github.com/spf13/cobra"
-	"k8s.io/klog/v2/klogr"
+	"k8s.io/klog/v2/textlogger"
```
Comment (fgiudici) [inline cmd/operator/main.go:21]: Not against it, but curious: why the switch from klogr to textlogger?

File: controllers/managedosimage_controller.go [hunk 2]
```diff
@@ -412,7 +412,7 @@ func (r *ManagedOSImageReconciler) mapImageToBundle(managedOSImage elementalv1.M
 	if managedOSImage.Namespace == fleetLocalNamespace {
 		bundle.Spec.Targets = []fleetv1.BundleTarget{{ClusterName: "local"}}
 	} else {
-		bundle.Spec.Targets = convertBundleTargets(managedOSImage.Spec.Targets)
+		bundle.Spec.Targets = managedOSImage.Spec.Targets
```
Comment (fgiudici) [inline controllers/managedosimage_controller.go:415]: 💚

**Review by fgiudici — APPROVED** [review]
great job 👏🏼

File: cmd/operator/main.go (hunk 1 above)
Comment (anmazzotti) [reply cmd/operator/main.go:21]: It's deprecated and the linter flagged it.

**Pull request merged**